test/helpers/xcm.ts

```diff
@@ -880,3 +887,262 @@ export const expectXcmEventMessage = async (context: DevModeContext, message: st
+      },
+    ],
+    weight_limit: {
+      refTime: 40_000_000_000n,
+      proofSize: 150_000n,
+    },
+    beneficiary: sovereignAccountOfSibling(context, paraId),
+  })
+    .withdraw_asset()
+    .buy_execution()
+    .push_any({
+      Transact: {
+        originKind: opts?.originKind ?? "Xcm",
+        requireWeightAtMost: {
+          refTime: 20_089_165_000n,
+          proofSize: 80_000n,
+        },
+        call: {
+          encoded: encodedCall,
+        },
+      },
+    })
+    .report_transact_status(
+      {
+        parents: 1,
+        interior: { X1: { Parachain: paraId } },
+      },
+      QUERY_ID
+    )
+    .refund_surplus()
+    .deposit_asset(1n, null, {
+      parents: 1,
+      interior: { X1: { Parachain: paraId } },
+    })
+    .as_v4();
+
+  const mockHrmpExistanceTx = context
+    .polkadotJs()
+    .tx.sudo.sudo(mockHrmpChannelExistanceTx(context, paraId, 1000, 102400, 102400));
+  await mockHrmpExistanceTx.signAndSend(alith);
+
+  // Send an XCM and create block to execute it
+  await injectHrmpMessage(context, paraId, {
+    type: "XcmVersionedXcm",
+    payload: xcmMessage,
+  } as RawXcmMessage);
+
+  const { block } = await context.createBlock();
+
+  const event = await expectEvent(context, block.hash as `0x${string}`, "Processed");
+  expect(context.polkadotJs().events.messageQueue.Processed.is(event)).to.be.true;
+  // Processed.success == true, to check that xcm message was processed successfully
+  expect(event.data[3].toJSON()).to.be.true;
+
+  const transactStatusMsg = (
+    await context.polkadotJs().query.parachainSystem.hrmpOutboundMessages()
+  )[0];
+
+  const transactStatusEncoded = "0x" + transactStatusMsg.data.toHex().slice(4);
+  const transactStatusDecoded = context
+    .polkadotJs()
+    .createType("XcmVersionedXcm", transactStatusEncoded) as XcmVersionedXcm;
+
+  let didSucceed = false;
+  let errorName: string | null = null;
+
+  const transactStatusQuery = transactStatusDecoded.asV4[0].asQueryResponse;
+  expect(transactStatusQuery.queryId.toNumber()).to.be.eq(QUERY_ID);
+  const dispatch = transactStatusQuery.response.asDispatchResult;
+  if (dispatch.isSuccess) {
+    didSucceed = true;
+  } else {
+    const error = dispatch.asError;
+    const dispatchError = context.polkadotJs().createType("DispatchError", error) as DispatchError;
+    if (dispatchError.isModule) {
+      const err = context.polkadotJs().registry.findMetaError({
+        index: dispatchError.asModule.index,
+        error: dispatchError.asModule.error,
+      });
+      errorName = err.name;
+    } else {
+      errorName = dispatchError.type;
+    }
+  }
+
+  if (!allowFailure) {
+    expect(errorName).to.be.null;
+    expect(didSucceed).to.be.true;
+  }
+
+  // this seems to fix an issue where we see SubscribeVersion instead of QueryResponse.
+  // a better fix would be set a correct XCM Version in pallet xcm for the parachain,
+  // so pallet xcm wouldn't need to send a SubscribeVersion message.
```

Comment:
This description is not clear, `SubscribeVersion` seems to be something mandatory.

Reply:
What do you mean by `SubscribeVersion ` being mandatory?

Reply:
I think it always needs to send a `SubscribeVersion`,  parachain safe xcm version can change, it is nothing hard-coded.